### PR TITLE
Make scoped instance IDisposable

### PR DESCRIPTION
**Pull request by yishaigalatzer**
Made the change to dispose content of `IScopedInstance` if the DI system disposes it.

Also some other minor cleanup on files I touched during this work.

CC / @divega @HaoK

### CODE REVIEW

File: src/Microsoft.Framework.DependencyInjection/ServiceDescriptor.cs [hunk 1]
```diff
@@ -57,18 +57,18 @@ private ServiceDescriptor(Type serviceType, LifecycleKind lifecycle)
         }
 
         /// <inheritdoc />
-        public LifecycleKind Lifecycle { get; private set; }
+        public LifecycleKind Lifecycle { get; }
```
Comment (yishaigalatzer) [inline src/Microsoft.Framework.DependencyInjection/ServiceDescriptor.cs]: C#6 cleanup

File: test/Microsoft.Framework.DependencyInjection.Tests/ScopingContainerTestBase.cs [hunk 2]
```diff
@@ -4,9 +4,9 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
```
Comment (yishaigalatzer) [inline test/Microsoft.Framework.DependencyInjection.Tests/ScopingContainerTestBase.cs]: ordering

**Comment by HaoK** [comment]
:shipit: looks good, do we want to consider adding a psuedo functional test that verifies that all the scoped instances actually get disposed when the service provider goes away?

**Comment by yishaigalatzer** [comment]
I was considering it. But didn't see functional tests for di, and didn't think this small change justifies it. Up to you

**Comment by yishaigalatzer** [comment]
Merged 8d29c5b8d6539249ca4c97da8c63083f6b53a9af

**Pull request closed without merging**